onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -23,6 +23,8 @@
 #include "core/providers/coreml/model/host_utils.h"
 #include "core/providers/coreml/shape_utils.h"
 
+#define HAS_GET_BYTES_WITH_HANDLER_API @available(macOS 12.3, iOS 15.4, *)
```

Comment:
do we need to introduce a macro if it's only used in one place?
maybe putting the `@available` inline and adding a comment about what it's guarding is good enough.